### ISSUE TITLE
[Lighthouse a11y] Format requirement missing for Birthday Form Fields

### ISSUE THREAD

**Issue by shreyachatterjee00** [civiform/civiform#12888]
Describe the bug
Issue: Issue: Format requirement missing for Birthday Form Fields (Day, Year)
 
Expectation: A visual and programatic label should be present to inform the user how to properly fill the form field.
 
No format expectations or requirements are present, which is important as the Edit Fields accept free text.
Format requirements are only present after putting in a date over 31 or year with 1 to 3 characters, and then selecting the "Continue" Button.

Location: Comprehensive Sample Program Form Screen 1

WCAG Success Criterion
3.3.2 Labels or Instructions

PC/JAWS
 1. Navigate to the "Day" or "Year" Form Fields by pressing Tab or JAWS Hotkey "E".
 
 iPhone/VO
 1. Navigate to the "Day" or "Year" Form Fields by swiping to the right.
 2. Move focus into the Text Field to edit by performing a double-tap with 1 finger.

Remediation: 
Convert the "Day" and "Year" free text Edit Fields to dropdown, spinner, or picker Form Fields with constraints.
 Limit day options to 1-31 depending on selected month. 
 Limit max and min year to reasonable years.
 
 or
 Add visual and programatic format instructions for "Day" and"Year".
 *Enter 1-31* or *Enter a 4 digit year e.g. 1996*
